LandSandBoat/server
# [cpp] Adds ability to modify the aoe radius of mobskills

**_I affirm:_**

## What does this pull request do?
This will add the ability to specify the AOE radius of the mobskill instead of defaulting it to the distance check.

Currently the radius is just defaulted to `m_Distance` for whatever reason, it just looks like it was never implemented. I have added a default of 0 with default checks inside the `getRadius` function for mobskills we do not have the info for and defaulted it to what we had in the codebase already.

This will simply give us the ability to modify the aoe radius as each mobskill is audited/implemented.


## Steps to test these changes
In order to test this will you need two characters and to test a skill that has a `mob_skill_aoe` of 2 and set its `mob_skill_aoe_radius` to the desired value.

I used `trebuchet` as my test skill as I was implementing it for Dynamis mobskill audit.
Run this query:
```
UPDATE `mob_skills` SET mob_skill_aoe = 2, mob_skill_aoe_radius = 12, mob_skill_distance = 16 WHERE mob_skill_id = 1636;
```
1) Make a party with your characters
2) Go to any ranger/ninja mob (I used a dynamis ninja)  `!gotoid 16945247`
3) Have char 1 aggro the mob and stand 15 away from the target
4) Have the Chare 2 stand less than 10 yards away from Char 1
5) `!exec target:useMobAbility(1636)` on the Hitman
You should see the AOE hit you from a distance of less than 10
6) Now step farther than 10 away from Char 1
7) `!exec target:useMobAbility(1636)` on the Hitman
You should see the AOE only hit Char 1 since you are too far away
